### PR TITLE
[Java] Kyma On-Prem Connectivity Workaround

### PR DESCRIPTION
**Pull request by Johannes-Schneider**
## What Has Changed?

This PR adds a workaround for the currently broken On-Premise connectivity from within the Kyma environment.

## Manual Checks?

- [x] Text adheres to the style guide (`vale docs/`)
  - Every sentence is on its own line
  - Headings use [title capitalization](https://capitalizemytitle.com/style/AP/#) (applies also to the sidebar and title of a document)
  - You followed [naming center](https://www.sapbrandtools.com/naming-center/#/dashboard) guidelines when referring to SAP products (e.g. SAP S/4HANA)
- [x] You checked your spelling and grammar (consider using Grammarly, see `CONTRIBUTING.md`)
- [x] You formatted all changed files with prettier (`npm run prettier`)
- [x] You tested if the documentation still builds (`npm run build`)
- [x] You verified all new and changed links still work (changing the `id` or name of a file can break links)
- [x] You have updated the [feature matrix](https://github.com/SAP/cloud-sdk/blob/main/docs/components/data/features.js) if you add documentation on a new feature or otherwise required.

### CODE REVIEW

File: docs-java/environments/sap-btp-kyma-environment.mdx [hunk 1]
```diff
@@ -20,6 +20,71 @@ While the Kyma runtime is improving over time, recent changes unfortunately rend
 We are currently re-evaluating and updating the information provided.
 Please follow the [official documentation](https://kyma-project.io/docs/kyma/latest/01-overview/).
 
+<details>
+  <summary>Workaround for On-Premise Connectivity</summary>
+
+Some of the most recent changes of the Kyma runtime environment render the SAP Cloud SDK's On-Premise connectivity unfunctional.
```
Comment (Johannes-Schneider) [inline docs-java/environments/sap-btp-kyma-environment.mdx:26]:
```suggestion
Recent changes of the Kyma runtime environment prevent the SAP Cloud SDK's On-Premise connectivity from working as intended.
```

File: docs-java/environments/sap-btp-kyma-environment.mdx [hunk 2]
```diff
@@ -20,6 +20,71 @@ While the Kyma runtime is improving over time, recent changes unfortunately rend
...
+        final Option<ProxyType> proxyType = requestContext.getDestination().get("ProxyType", ProxyType.class);
+        if( proxyType.getOrElse(ProxyType.INTERNET) != ProxyType.ON_PREMISE ) {
+            return Collections.emptyList();
+        }
+
+        final Option<AuthenticationType> authentication =
+            requestContext.getDestination().get("authentication", AuthenticationType.class);
+        if( authentication
+            .getOrElse(AuthenticationType.NO_AUTHENTICATION) != AuthenticationType.PRINCIPAL_PROPAGATION ) {
+            return Collections.emptyList();
```
Comment (MatKuhr) [inline docs-java/environments/sap-btp-kyma-environment.mdx:48]: Why don’t we cover principal propagation? Is it specific to the untrusted mode?

File: docs-java/environments/sap-btp-kyma-environment.mdx [hunk 3]
```diff
@@ -20,6 +20,71 @@ While the Kyma runtime is improving over time, recent changes unfortunately rend
...
+        }
+
+        final Try<String> maybeBearerToken =
+            AuthTokenAccessor
+                .tryGetCurrentToken()
+                .map(token -> String.format("Bearer %s", token.getJwt().getToken()));
+
+        if( maybeBearerToken.isEmpty() ) {
+            return Collections.emptyList();
```
Comment (MatKuhr) [inline docs-java/environments/sap-btp-kyma-environment.mdx:57]: At this point a warning might make sense, not sure if there is a scenario where the onprem call afterwards could still succeed

**Review by MatKuhr — APPROVED** [review]
Lgtm; assuming the formatting works out

File: docs-java/environments/sap-btp-kyma-environment.mdx [hunk 4]
```diff
@@ -20,6 +20,71 @@ While the Kyma runtime is improving over time, recent changes unfortunately rend
...
+    {
+        final Option<ProxyType> proxyType = requestContext.getDestination().get("ProxyType", ProxyType.class);
+        if( proxyType.getOrElse(ProxyType.INTERNET) != ProxyType.ON_PREMISE ) {
+            return Collections.emptyList();
+        }
+
+        final Option<AuthenticationType> authentication =
+            requestContext.getDestination().get("authentication", AuthenticationType.class);
+        if( authentication
+            .getOrElse(AuthenticationType.NO_AUTHENTICATION) != AuthenticationType.PRINCIPAL_PROPAGATION ) {
```
Comment (newtork) [inline docs-java/environments/sap-btp-kyma-environment.mdx:47]: Maybe slightly less code...?
```suggestion
        if( !authentication.contains(AuthenticationType.PRINCIPAL_PROPAGATION) ) {
```

**Review by newtork — CHANGES_REQUESTED** [review]
Please check whether this could be a "workaround" instead
```
static{
  PrincipalPropagationStrategy.setDefaultStrategy(PrincipalPropagationStrategy.COMPATIBILITY);
}
```

IF not, please dismiss my review.

**Comment by Johannes-Schneider** [comment]
Converted to a draft since we are evaluating whether the suggestion from @newtork is also working.

**Comment by Johannes-Schneider** [comment]
The customer didn't reply yet, so we are assuming that the suggested workaround is good enough for the moment (as it has been tested).

**Pull request merged**